hack/tools/pkg/janitor/janitor.go

```diff
@@ -197,6 +207,118 @@ func (s *Janitor) deleteVSphereVMs(ctx context.Context, folder string) error {
 	return nil
 }
 
+// DeleteCNSVolumes deletes all volumes from tests.
+func (s *Janitor) DeleteCNSVolumes(ctx context.Context, boskosResourceName string) error {
+	log := ctrl.LoggerFrom(ctx).WithName("volumes")
+	ctx = ctrl.LoggerInto(ctx, log)
+
+	log.Info("Deleting volumes")
+
+	type cnsVolumeToDelete struct {
+		volumeID     cnstypes.CnsVolumeId
+		pvcName      string
+		pvcNamespace string
+	}
+	volumesToDelete := []cnsVolumeToDelete{}
+
+	queryFilter := cnstypes.CnsQueryFilter{
+		Labels: []types.KeyValue{
+			{
+				Key:   boskosResourceLabel,
+				Value: boskosResourceName,
+			},
+		},
+	}
+
+	for {
+		res, err := s.vSphereClients.CNS.QueryVolume(ctx, queryFilter)
+		if err != nil {
+			return err
+		}
+
+		for _, volume := range res.Volumes {
+			var pvcMetadata *cnstypes.CnsKubernetesEntityMetadata
+			for _, meta := range volume.Metadata.EntityMetadata {
+				k8sMetadata, ok := meta.(*cnstypes.CnsKubernetesEntityMetadata)
+				if !ok {
+					continue
+				}
+				if k8sMetadata.EntityType != string(cnstypes.CnsKubernetesEntityTypePVC) {
+					continue
+				}
+				pvcMetadata = k8sMetadata
+			}
+
+			if pvcMetadata == nil {
+				// Ignoring non-PVC volumes.
+				continue
+			}
+
+			var matchesBoskosResourcename bool
+			// Check again that the volume has a matching label.
+			for _, v := range pvcMetadata.Labels {
+				if v.Key != boskosResourceLabel {
+					continue
+				}
+				if v.Value != boskosResourceName {
+					continue
+				}
+				matchesBoskosResourcename = true
+			}
+
+			// Ignore not matching volume.
+			if !matchesBoskosResourcename {
+				continue
+			}
+
+			volumesToDelete = append(volumesToDelete, cnsVolumeToDelete{
+				volumeID:     volume.VolumeId,
+				pvcName:      pvcMetadata.EntityName,
+				pvcNamespace: pvcMetadata.Namespace,
+			})
+		}
+
+		if res.Cursor.Offset == res.Cursor.TotalRecords || len(res.Volumes) == 0 {
+			break
+		}
+
+		queryFilter.Cursor = &res.Cursor
+	}
+
+	if len(volumesToDelete) == 0 {
+		log.Info("No CNS Volumes to delete")
+		return nil
+	}
+
+	deleteTasks := []*object.Task{}
+	for _, volume := range volumesToDelete {
+		log := log.WithValues("volumeID", volume.volumeID, "pvcName", volume.pvcName, "pvcNamespace", volume.pvcNamespace)
+
+		log.Info("Deleting CNS Volume in vSphere", "volumeID", volume.volumeID)
```

Comment:
```suggestion
		log.Info("Deleting CNS Volume in vSphere")
```
added to the logger above